File: Editorial/0000-0100/0003_Longest_Substring_Without_Repeating_Characters/JavaScript/1_Brute_Force_TLE.js
var lengthOfLongestSubstring = function (s) {
    const n = s.length;

    let res = 0;
    for (let i = 0; i < n; i++) {
        for (let j = i; j < n; j++) {
            if (checkRepetition(s, i, j)) {
                res = Math.max(res, j - i + 1);
            }
        }
    }

    function checkRepetition(s, start, end) {
        const chars = new Set();

        for (let i = start; i <= end; i++) {
            const c = s[i];
            if (chars.has(c)) {
                return false;
            }
            chars.add(c);
        }

        return true;
    }

    return res;
};

File: Editorial/0000-0100/0003_Longest_Substring_Without_Repeating_Characters/C/1_Brute_Force_TLE.c
bool checkRepetition(char* s, int start, int end) {
    bool chars[128] = {0};
    for (int i = start; i <= end; i++) {
        char c = s[i];
        if (chars[c]) {
            return false;
        }
        chars[c] = true;
    }
    return true;
}

int max(int a, int b) { return (a > b) ? a : b; }

int lengthOfLongestSubstring(char* s) {
    int n = strlen(s);
    int res = 0;
    for (int i = 0; i < n; i++) {
        for (int j = i; j < n; j++) {
            if (checkRepetition(s, i, j)) {
                res = max(res, j - i + 1);
            }
        }
    }
    return res;
}
